### ISSUE TITLE
[HOLD for payment 2023-02-01]  Console error appears when clicking `Save&Continue` without selecting the bank account after successful plaid flow.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#13598]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Select Connect bank account for any workspace
2. Choose connect online with plaid option
3. Complete the plaid selection flow by selecting multiple accounts
4. After navigating to our app try to click Save and continue button without selecting account from the dropdown
5. Observe the console as it will throw an error

## Expected Result: 
Should show the related error(Choose the account plz) and console error shouldn’t be there

## Actual Result: 
Console error(Cannot read properties of undefined (reading ‘routingNumber’))

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platform:

Where is this issue occurring?

- Web

**Version Number:** 1.2.39-0
**Reproducible in staging?:** y
**Reproducible in production?:** Could not check
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/207707484-0e63be73-34f5-49ac-8738-0a80b8f93a5e.mp4

https://user-images.githubusercontent.com/43996225/207707514-ebb14884-cf4f-423f-9f65-6484446cb667.mov

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @Pujan92 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1671026877428529

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by thienlnam** [comment]
Nabbing this and taking it internal to complete - you shouldn't be able to hit Save & Continue without selecting the bank account

**Comment by thienlnam** [comment]
I've been digging into this and found out why this happens

When we load that page, we get the selectedPlaidAccountID from https://github.com/Expensify/App/blob/91f7bac522377034b1380df5fce52b6095a3a43c/src/pages/ReimbursementAccount/BankAccountPlaidStep.js#L65-L65 

the reimbursement account draft

https://github.com/Expensify/App/blob/91f7bac522377034b1380df5fce52b6095a3a43c/src/libs/ReimbursementAccountUtils.js#L24-L27

Basically, if we've already gotten to this step and select an account, we'll have stored the bank account in that onyx key.  So when you come back to it, it grabs that selectedAccountID from the old bank account

**Comment by thienlnam** [comment]
An easy way to reproduce is 

1. Create a workspace, add a new bank account and select a bank account on that screen
2. Exit out, and then go through the process again connecting with plaid and notice that even when you are shown 'Choose an account', the button is visible.

You can just fix it by signing out which will clear your onyx data.

This problem only happens when you decide to go back and reconnect to another bank account with plaid.

**Comment by nkuoch** [comment]
Should be fixed by https://github.com/Expensify/App/pull/13668

**Comment by flaviadefaria** [comment]
As a note, I'm going to be OOO from Dec 23rd to January 9th. In case this needs to be paid before the week of January 9th then please reassign.

**Comment by flaviadefaria** [comment]
@nkuoch and @thienlnam can I close this GH given this will be fixed by https://github.com/Expensify/App/pull/13236?

**Comment by nkuoch** [comment]
Let's keep it open, and it will close once the PR is deployed. Meanwhile, let's just switch it to weekly

**Comment by thienlnam** [comment]
Linked PR https://github.com/Expensify/App/pull/13236 is in review

**Comment by flaviadefaria** [comment]
Switching this back to daily so that I don't miss payment.

**Comment by flaviadefaria** [comment]
Chill Melvin, I'm only paying them on the 1st.

**Comment by flaviadefaria** [comment]
I'm going to issue payment to @Pujan92 for reporting the bug. Once that's done are we good to close this @nkuoch and @thienlnam?

**Comment by flaviadefaria** [comment]
Payment has been made to @Pujan92. I'm going to close this, but please reopen it in case I missed something.
